core/solver/gmres.cpp

```diff
@@ -112,6 +142,156 @@ struct help_compute_norm {
     }
 };
 
+namespace {
```

Comment:
Not sure if it's technically necessary. I thought perhaps it would be preferred to keep the helper functions contained to this file; I noticed it was used for some helper functions in multigrid. But, now, as I look at more files, it seems multigrid is an outlier? I knew usually an anonymous namespace is used for registering the kernels, but in multigrid it's not, and then it's used for some other functions...are there some other rules for use of anonymous namespace in Ginkgo?

Anyway, I will remove it from the gmres helper functions.